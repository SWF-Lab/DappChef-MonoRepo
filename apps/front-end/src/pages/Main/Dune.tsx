import { Container } from "./styles"

//front end
import Stack from "@mui/material/Stack"
import { makeStyles } from "@material-ui/core/styles"
import * as React from "react"
import Paper from "@mui/material/Paper"
import Box from "@material-ui/core/Box"
import Grid from "@mui/material/Grid"
import { styled } from "@mui/material/styles"
import Typography from "@mui/material/Typography"
import bg from "../../components/Img/Dune6.png"

const useStyles = makeStyles((theme) => ({
  root: {
    width: "80vw",
    height: "93vh",
    position: "relative",
    display: "flex",
    // background: "#0F0B18",
    alignItems: "center",
    justifyContent: "center",
    objectFit: "contain",
    padding: 30
    // "& video": {
    //   objectFit: "cover"
    // }
  },
  overlay: {
    position: "absolute",
    top: 0,
    left: 0,
    width: "100%",
    height: "100%"

    // backgroundColor:","
  },
  title: {
    paddingBottom: theme.spacing(4)
  }
}))

export const Dune = () => {
  const classes = useStyles()
  return (
    <>
      <Container
        maxWidth={false}
        style={{ flexDirection: "row", overflow: "visible" }}
      >
        <div
          className={classes.root}
          style={{
            backgroundImage: `url(${bg})`,
            backgroundSize: "contain",
            backgroundRepeat: "no-repeat",
            backgroundPosition: "center"
            // backgroundAttachment: "fixed"
          }}
        >
          {/* <div className={classes.overlay}> */}
          <Grid
            height="80vh"
            width="75%"
            display="flex"
            flexDirection="column"
            justifyContent="space-between"
            alignItems="flex-start"
            // alignItems="center"
            // spacing={7}
            // sx={{ p: 15 }}
          >
            <Typography
              variant="h4"
              align="center"
              color="white"
              component="p"
              sx={{
                typography: { lg: "h4", md: "body1", sm: "caption" }
                //   pb:3
              }}
            >
              Find your Dune data here...
            </Typography>
            <Stack
              direction="row"
              justifyContent="space-around"
              alignItems="center"
              spacing={5}
              // sx={{mt:5}}
            >
              <Paper
                style={{
                  width: { lg: "220px", md: "150px", sm: "100px" },
                  height: { lg: "220px", md: "150px", sm: "100px" },
                  backgroundColor: "white"
                }}
                sx={{
                  m: 1,

                  borderRadius: "15px",
                  border: "10px solid #D9D9D9"
                }}
              >
                <iframe
                  src="https://dune.com/embeds/2022989/3349012"
                  height="100%"
                  width="100%"
                  style={{ border: "none" }}
                />
              </Paper>
              <Paper
                style={{
                  width: "220px",
                  height: "220px",
                  backgroundColor: "white"
                }}
                sx={{
                  m: 1,

                  borderRadius: "15px",
                  border: "10px solid #D9D9D9"
                }}
              >
                <iframe
                  src="https://dune.com/embeds/2023032/3349086"
                  height="100%"
                  width="100%"
                  style={{ border: "none" }}
                />
              </Paper>
              <Paper
                style={{
                  width: "220px",
                  height: "220px",
                  backgroundColor: "white"
                }}
                sx={{
                  m: 1,

                  borderRadius: "15px",
                  border: "10px solid #D9D9D9"
                }}
              >
                <iframe
                  src="https://dune.com/embeds/2023164/3349291"
                  height="100%"
                  width="100%"
                  style={{ border: "none" }}
                />
              </Paper>
            </Stack>
            <Stack
              flexDirection="row"
              justifyContent="space-around"
              alignItems="center"
              //   spacing={5}
            >
              <Paper
                style={{
                  width: "300px",
                  height: "230px",
                  backgroundColor: "white"
                }}
                sx={{
                  ml: { xl: 5, md: 0 },
                  borderRadius: "15px",
                  border: "10px solid #D9D9D9"
                }}
              >
                <iframe
                  src="https://dune.com/embeds/2023125/3349229"
                  height="100%"
                  width="100%"
                  style={{ border: "none" }}
                />
              </Paper>
              <Paper
                style={{
                  width: "500px",
                  height: "230px",
                  backgroundColor: "white"
                }}
                sx={{
                  ml: 5,
                  borderRadius: "15px",
                  border: "10px solid #D9D9D9"
                }}
              >
                <iframe
                  src="https://dune.com/embeds/2023176/3349282"
                  height="100%"
                  width="100%"
                  style={{ border: "none" }}
                />
              </Paper>
            </Stack>
          </Grid>
          {/* </div> */}
        </div>
      </Container>
    </>
  )
}
