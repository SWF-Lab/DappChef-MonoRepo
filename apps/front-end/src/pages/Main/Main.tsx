import { useState, useEffect, useRef } from "react"
import { TextExample } from "src/components/TextExample/TextExample"
import { useNavigate } from "react-router-dom"
import { ethers } from "ethers"
import { Container } from "./styles"
import { ProblemList } from "../ProblemsInterface/ProblemList"
import { ResponsiveAppBar } from "../../components/Appbar"
import { useHook } from "../useHooks"
import { TotalMarquee } from "../../components/Murquee"
import { Footer } from "../../components/Footer"
import Grid from "@mui/material/Grid"
import styled from "styled-components"
import Box from "@material-ui/core/Box"
import ReactPlayer from "react-player"
import BeforeLoginVideo from "../../components/Img/DappChef_v3_BeforeLogin.mp4"
import LoginVideo from "../../components/Img/DappChef_v8_login.mp4"
import start from "../../components/Img/Start.png"
import { makeStyles } from "@material-ui/core/styles"
import Ads from "../../components/Ads.tsx"
import { Ad } from "../../components/Ad.tsx"
import { Dune } from "./Dune.tsx"

const useStyles = makeStyles((theme) => ({
  root: {
    width: "100%",
    height: "100%",
    position: "relative",
    display: "flex",
    backgroundColor: "#0F0B18",
    alignItems: "center",
    justifyContent: "center",
    "& video": {
      objectFit: "cover"
    }
  },
  overlay: {
    position: "absolute",
    top: 0,
    left: 0,
    width: "100%",
    height: "100%"
    // backgroundColor:","
  },
  title: {
    paddingBottom: theme.spacing(4)
  }
}))

export const Main = () => {
  const { account } = useHook()
  const classes = useStyles()
  const imageStyle = {
    cursor: "pointer"
  }

  /*要再整理 */
  const scrolltoProblem = () => {
    const width = window.innerWidth
    const height = window.innerHeight

    let scrollTop = height * 2.18
    console.log(width, height, scrollTop)
    // if (width >= 1280) {
    //   scrollTop = height*2;
    // } else if (width >= 1024) {
    //   scrollTop = 1450;
    // }  else {
    //   scrollTop = 1000;
    // }
    window.scrollTo({ top: scrollTop, behavior: "smooth" })
  }

  const newwidth = () => {
    let widthnow = window.innerWidth
    console.log(widthnow)
    return widthnow
  }

  return (
    <>
      <ResponsiveAppBar />
      <main
        style={{
          background: "#0F0B18",
          height: "100%"
          // overflow: "auto"
        }}
      >
        {/* <Container> */}
        {account ? (
          <section className={classes.root}>
            <div style={{ width: "100%", backgroundColor: "red" }}>
              <ReactPlayer
                url={LoginVideo}
                playing
                loop
                muted
                style={{ objectFit: "cover", width: "100%", height: "100%" }}
                width="100%"
                height="100%"
              />
            </div>
            <div className={classes.overlay}>
              <Box
                height="100%"
                display="flex"
                flexDirection="row"
                justifyContent="center"
                alignItems="flex-end"
                sx={{ pb: 15 }}
              >
                <img
                  src={start}
                  width={"21%"}
                  style={imageStyle}
                  alt="start"
                  onClick={scrolltoProblem}
                />
              </Box>
            </div>
          </section>
        ) : (
          <section className={classes.root}>
            {/* <div style={{ width: '100%', height: '100vh' }}> */}
            <ReactPlayer
              url={BeforeLoginVideo}
              playing
              muted
              width="100%"
              height="100%"
            />
            {/* </div> */}
            <div className={classes.overlay}>
              <Box
                height="100%"
                display="flex"
                flexDirection="row"
                justifyContent="flex-end"
                alignItems="flex-end"
                sx={{ px: 20, pb: 15 }}
              >
                <img
                  src={start}
                  width={"23%"}
                  style={imageStyle}
                  alt="start"
                  onClick={scrolltoProblem}
                />
              </Box>
            </div>
          </section>
        )}
        <Ads />

        <Dune />
        <Grid>
          <ProblemList />
        </Grid>
        <TotalMarquee />
      </main>
      <Footer />
    </>
  )
}
