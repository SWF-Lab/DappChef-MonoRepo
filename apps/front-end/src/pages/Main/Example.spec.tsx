import { render, screen } from "@testing-library/react"
import renderer from "react-test-renderer"
import { Main } from "src/pages/Main/Main"

describe("Example Page", () => {
  it("should render correcty", () => {
    const component = renderer.create(<Main />)

    expect(component).toMatchSnapshot()
  })
  it("should be able to render Example text", () => {
    render(<Main />)

    expect(screen.getByText(/Main/i)).toBeInTheDocument()
  })
})
