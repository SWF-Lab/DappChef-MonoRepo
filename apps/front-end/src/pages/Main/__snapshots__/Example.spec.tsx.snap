// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example Page should render correcty 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 100vh;
}

<div
  className="c0"
>
  <h1>
    Example
  </h1>
</div>
`;
